code/modules/client/client procs.dm

```diff
@@ -431,11 +438,104 @@
 	var/DBQuery/query_accesslog = dbcon.NewQuery("INSERT INTO `[format_table_name("connection_log")]`(`id`,`datetime`,`serverip`,`ckey`,`ip`,`computerid`) VALUES(null,Now(),'[serverip]','[ckey]','[sql_ip]','[sql_computerid]');")
 	query_accesslog.Execute()
 
-
 #undef TOPIC_SPAM_DELAY
 #undef UPLOAD_LIMIT
 #undef MIN_CLIENT_VERSION
 
+// Returns true if a randomizer is being used
+/client/proc/check_randomizer()
+	. = FALSE
+	if(!config.check_randomizer)
+		return
+	// Stash o' ckeys
+	var/static/cidcheck = list()
+	// Ckeys that failed the test, stored to send acceptance messages only for atoners
+	var/static/cidcheck_failedckeys = list()
+
+	var/oldcid = cidcheck[ckey]
+	if(!oldcid)
+		// We don't have their cached CID on hand, so ask the DB
+		var/sql_ckey = sanitizeSQL(ckey)
+		var/DBQuery/query_cidcheck = dbcon.NewQuery("SELECT computerid FROM [format_table_name("player")] WHERE ckey = '[sql_ckey]'")
+		query_cidcheck.Execute()
+
+		var/lastcid = computer_
+		if(query_cidcheck.NextRow())
+			lastcid = query_cidcheck.item[1]
+
+		if(computer_id != lastcid)
+			// Their current CID does not match what the DB says - OFF WITH THEIR HEAD
+			cidcheck[ckey] = computer_id
+
+			// Disable the reconnect button to force a CID change
+			winset(src, "reconnectbutton", "is-disable=true")
+
+			log_access("Failed Login: [key] [computer_id] [address] - CID randomizer check")
+
+			var/url = winget(src, null, "url")
+			// Javascript trick to make them reconnect in a new window
+			src << browse("<a id='link' href=byond://[url]>byond://[url]</a>\
+			<script type='text/javascript'>\
+				document.getElementById(\"link\").click();\
+				window.location=\"byond://winset?command=.quit\"\
+			</script>", "border=0;titlebar=0;size=1x1")
+			sleep(10) // Since browse is non-instant, and kinda async
+
+			// If the above didn't work, pretend that there's just a read error (slimy)
+			to_chat("<pre class=\"system system\">Network connection shutting down due to read error ;).</pre>")
+			del(src)
+			return TRUE
+	else
+		// We DO have their cached CID handy - compare it, now
+		if(oldcid != computer_id)
+			// Change detected, they are randomizing
+			cidcheck -= ckey	// To allow them to try again after removing CID randomization
+
+			to_chat("<span class='userdanger'>Connection Error:</span>")
+			to_chat("<span class='danger'>Invalid ComputerID(spoofed). Please remove the ComputerID spoofer from your BYOND installation and try again.</span>")
+
+			if(!cidcheck_failedckeys[ckey])
+				message_admins("<span class='adminnotice'>[key_name(src)] has been detected as using a CID randomizer. Connection rejected.</span>")
+				send2irc_adminless_only(config.cidrandomizer_irc, "[key_name(src)] has been detected as using a CID randomizer. Connection rejected.")
```

Comment:
as my other comment apparently died, don't use `_adminless_only` please, this should be recorded regardless